elastic/kibana
# [Remote clusters] Migrate client-side code out of legacy

This PR _mostly_ completes the NP migration for the `remote_clusters` plugin.

Continuation of https://github.com/elastic/kibana/pull/56781.

### What's left
* Once the index_management plugin is fully migrated, it should be a required plugin for remote_clusters
* Once CCR is migrated, the legacy config should be removed from remote_clusters

### How to test
You will need to set up two clusters in order to test remote clusters. For example:

    yarn es snapshot
    yarn es snapshot -E cluster.name="my_cluster" -E http.port=9201 -E path.data=/tmp/es (in another tab)

* Test all actions behave as expected (create, edit, delete remote clusters)
* Test integration and redirect behavior with CCR (e.g., if you create a remote cluster from the CCR flow, you should be redirected back to CCR)
* Confirm remote clusters is disabled if `xpack.remote_clusters.enabled`  is `false`
* Test error handling (e.g., create a cluster with the same name as an existing one)
